README.md

```diff
@@ -1,14 +1,67 @@
-# Compass
+![Compass logo](https://raw.githubusercontent.com/hyperoslo/Compass/update/readme/Images/logo_v1.png)
 
 [![CI Status](http://img.shields.io/travis/hyperoslo/Compass.svg?style=flat)](https://travis-ci.org/hyperoslo/Compass)
 [![Version](https://img.shields.io/cocoapods/v/Compass.svg?style=flat)](http://cocoadocs.org/docsets/Compass)
 [![License](https://img.shields.io/cocoapods/l/Compass.svg?style=flat)](http://cocoadocs.org/docsets/Compass)
 [![Platform](https://img.shields.io/cocoapods/p/Compass.svg?style=flat)](http://cocoadocs.org/docsets/Compass)
 
-## Usage
+Compass helps you setup a central navigation system for your application.
+This has many benefits, one of them being that controllers can now be
+decoupled, meaning that the list that presents the detail no longer knows 
+about what its presenting. Controllers become agnostic and view stay 
+stupid. The user experience stays the same but the logic and separation of
+concerns become clearer. The outcome is that your application will become 
+more modular by default. Anything could potentially be displayed from 
+anywhere but do tread safe, with great power comes great responsibility.
 
+## Setup
+
+#### Step 1
+First you need to register a URL scheme for your application
+<img src="https://raw.githubusercontent.com/hyperoslo/Compass/update/readme/Images/setup-url-scheme.png"> 
+
+#### Step 2
+Now you need to configure Compass to use that URL scheme, a good place
+to do this is in your `AppDelegate`
+```swift
+func application(application: UIApplication, 
+  didFinishLaunchingWithOptions launchOptions: [NSObject: AnyObject]?) -> Bool {
+    Compass.scheme = "compass"
+    return true
+}
+```
+#### Step 3
+Configure your application routes
+```swift
+func application(application: UIApplication, 
+  didFinishLaunchingWithOptions launchOptions: [NSObject: AnyObject]?) -> Bool {
+    Compass.scheme = "compass"
+    Compass.routes = ["profile:{username}", "login:{username}", "logout"]
+    return true
+}
+```
+#### Step 4
+Setup your application to response to the URLs, this can be done in the `AppDelegate` but its up to you to find a more suitable place for it depending on the size of your implementation.
 ```swift
-<API>
+func application(app: UIApplication, 
```

Comment:
I think we have to mention that this func should be used to open a view from outside, but for internal usage it's better to have some global/static function, not to invoke `applicationDidBecomeActive` every time.